training/unit-testing/go-lang/README.md

```diff
@@ -49,75 +31,81 @@ In your terminal:
  go get golang.org/x/tools/cmd/cover
 ```
 
-### Quick Reference
+## Quick Reference
 The following are useful links:-
-- [GoConvey on GitHub](https://github.com/smartystreets/goconvey)
-- [GoConvey documentation](https://godoc.org/github.com/smartystreets/goconvey/convey)
+- [GoConvey on GitHub](https://github.com/smartystreets/goconvey) - walks through the GoConvey setup, introduced the Web UI, and has downloadable examples.
+- [GoConvey documentation](https://godoc.org/github.com/smartystreets/goconvey/convey) - details clearly the GoConvey functions and assertions available.
 
-### Potential Pitfalls
+## Potential Pitfalls
 
 #### Order of Execution
-GoConvey's execution path is a little quirky so have a look at the example below and the resources.
+GoConvey's execution path is a little quirky so have a look at the example below and the linked resources.
 - [Downloadable order of execution example](#orderexecution/execution_order_example_test.go)
 - [GoConvey explanation](https://github.com/smartystreets/goconvey/wiki/Execution-order)
 
 #### Assertions
-- When comparing objects in your tests you can use [GoConvey's Assertions](/assertions/README.md). 
+When comparing objects in your tests you can use [GoConvey's Assertions](/assertions/README.md), however when comparing complex `structs` such as `errors` can be troublesome.
```

Comment:
errors are a built-in type but under the hood represent an interface to allow users to create their own errors - this can make it harder to test as you may want to check the stringified error value or if you have made a custom error which contains other functions or values you would need to caste the error to your custom error.

Either way this is easy, average or difficult depending on your knowledge of go.

Basically I would not put an opinion of the difficulty in the documentation as it is subjective

Reply:
Instead maybe rephrase this that we can use DeepEqual to assist with comparing structs

Reply:
Yip that makes sense.  Thanks Nathan.